homeassistant/components/hisense_aehw4a1/config_flow.py

```diff
@@ -0,0 +1,22 @@
+"""Config flow for Hisense AEH-W4A1 integration."""
+import logging
+
+from pyaehw4a1.aehw4a1 import AehW4a1
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_flow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(hass):
+    """Return if there are devices that can be discovered."""
+    aehw4a1_ip_addresses = await AehW4a1().discovery()
+    return len(aehw4a1_ip_addresses) > 0
+
+
+config_entry_flow.register_discovery_flow(
```

Comment:
Please also add an import step that checks the connection to the ip address provided by config yaml.

Reply:
Here I'm stuck... I implemented a new function in library to check for presence:
`AehW4a1().check()` that return True or custom exception ConnectionError

But I don't know how to implement the import step

Reply:
See eg abode config flow:
https://github.com/home-assistant/home-assistant/blob/0954f7d3bef313686686db3ee539086ee441fc1e/homeassistant/components/abode/config_flow.py#L76-L82

Instead of a user step you can add a method that uses the library check.

Reply:
I don't know why import_config is always None

```
@config_entries.HANDLERS.register(DOMAIN)
class AWSFlowHandler(config_entries.ConfigFlow):
    """Handle a config flow."""

    VERSION = 1
    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL

    async def async_step_import(self, import_config):
        """Import a config entry from configuration.yaml."""
        if self._async_current_entries():
            _LOGGER.warning("Only one configuration of Hisense AEH-W4A1 is allowed.")
            return self.async_abort(reason="single_instance_allowed")

        if import_config is not None:
```

Reply:
Pass the domain config as data keyword argument when starting the config flow, via import, in the component.

Reply:
Ok, I did it and push my mod, but it is not working as intended. If I comment out the auto discovery part from config_flow.py HA raises my exception for check failed, but if I leave the code uncommented the autodiscovery gain priority over data on configuration.yaml

Reply:
Yeah we need to remove the discovery helper registration here and implement the correct discovery step in the same flow as the import step. There can only be one flow registered per domain.

Sorry about not being clear.

Reply:
You can copy the fundamental parts from here:
https://github.com/home-assistant/home-assistant/blob/206547a5d8abbcf7bc0463d312cc9f68cdcfce07/homeassistant/helpers/config_entry_flow.py#L29-L61

Reply:
So the entry point for the user in the flow will be the user step if configured via GUI. We won't implement a step called discovery.

Reply:
I think I've changed my mind, sorry.

There's an easier way that would let us keep the existing discovery registration. Instead of implementing an import step in a new flow, we check the ip address in the component setup function. Only if the ip address is ok, do we import the config and init the flow.

Reply:
Very well, much easier for me =) and thanks to you I finally understood how the config flow works! The development guides didn't help me much...

Reply:
I insert check in setup function, but I have a few questions:
If an ip address does not pass the test, should I exit the setup procedure or complete it for valid ip's?
What is the correct way to exit the function in these cases?

Reply:
it doesn't matter, I read below that the check must be inserted in __init__ discarding the addresses that do not pass it